### PR TITLE
Fasttext component

### PR DESCRIPTION
**Pull request by yfletberliac**
**Proposed changes**:
- ...

**Status (please check what you already did)**:
- [ ] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

File: rasa_nlu/classifiers/fasttext_intent_classifier.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    """Intent classifier using the sklearn framework
+    Here are the references for later use and specs adds
+    https://github.com/facebookresearch/fastText/blob/master/python/fastText/FastText.py"""
+
+
+    name = "intent_classifier_fasttext"
+
+    provides = ["intent", "intent_ranking"]
+
+    # requires = ["text_features"]
```
Comment (pocman) [inline rasa_nlu/classifiers/fasttext_intent_classifier.py]: Remove this line

File: rasa_nlu/classifiers/fasttext_intent_classifier.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+    def train(self, training_data, config, **kwargs):
+        # type: (TrainingData, RasaNLUConfig, **Any) -> None
+        """Train the intent classifier on a data set.
+
+        :param num_threads: number of threads used during training time"""
+
+        classifier_file = "projects/default/model.ftz"
+
+        # self.clf = load_model(classifier_file)
```
Comment (pocman) [inline rasa_nlu/classifiers/fasttext_intent_classifier.py]: Remove this line

File: rasa_nlu/classifiers/fasttext_intent_classifier.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        }
+
+
+    @classmethod
+    def load(cls, model_dir=None, model_metadata=None, cached_component=None, **kwargs):
+        # type: (Text, Metadata, Optional[Component], **Any) -> SklearnIntentClassifier
+        import cloudpickle
+        from fastText import load_model
+
+        model_file = "projects/default/model.ftz"
```
Comment (pocman) [inline rasa_nlu/classifiers/fasttext_intent_classifier.py]: This should probably be extracted from the model_metadata.

File: rasa_nlu/classifiers/fasttext_intent_classifier.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+        return [x.replace('__label__', '') for x in labels]
+
+    def train(self, training_data, config, **kwargs):
+        # type: (TrainingData, RasaNLUConfig, **Any) -> None
+        """Train the intent classifier on a data set.
+
+        :param num_threads: number of threads used during training time"""
+
+        classifier_file = "projects/default/model.ftz"
```
Comment (pocman) [inline rasa_nlu/classifiers/fasttext_intent_classifier.py]: This should be retrieved from the config and stored in `model_metadata`.

File: rasa_nlu/classifiers/fasttext_intent_classifier.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        model_file = "projects/default/model.ftz"
+
+        if model_dir and model_metadata.get("intent_classifier_fasttext"):
+            classifier_file = os.path.join(model_dir, model_metadata.get("intent_classifier_fasttext"))
+
+            with io.open(classifier_file, 'rb') as f:  # pragma: no test
+                if PY3:
+                    return cloudpickle.load(f, encoding="latin-1")
+                else:
+                    a = cloudpickle.load(f)
```
Comment (pocman) [inline rasa_nlu/classifiers/fasttext_intent_classifier.py]: Rename variable and add comment to explain why you don't pkl the clf.

File: rasa_nlu/server.py [hunk 6]
```diff
@@ -246,5 +246,6 @@ def evaluate(self, request):
     rasa_nlu_config = RasaNLUConfig(
             cmdline_args.get("config"), os.environ, cmdline_args)
     rasa = RasaNLU(rasa_nlu_config)
+    logger.info(rasa)
```
Comment (pocman) [inline rasa_nlu/server.py]: Remove this line

File: sample_configs/config_fasttext.json [hunk 7]
```diff
@@ -0,0 +1,5 @@
+{
+  "pipeline": "fasttext",
+  "path" : "./projects",
+  "data" : "./data/examples/rasa/demo-rasa.json"
```
Comment (pocman) [inline sample_configs/config_fasttext.json]: You should add "***/model.ftz" here

File: src/pip-delete-this-directory.txt [hunk 8]
```diff
@@ -0,0 +1,5 @@
+This file is placed here by pip to indicate the source was put
```
Comment (pocman) [inline src/pip-delete-this-directory.txt]: Remove this file

**Review by pocman — CHANGES_REQUESTED** [review]

File: rasa_nlu/classifiers/fasttext_intent_classifier.py (hunk 4 above)
Comment (yfletberliac) [reply rasa_nlu/classifiers/fasttext_intent_classifier.py]: not in the standard (see ```rasa_nlu/classifiers/sklearn_intent_classifier.py```) - the *persist* method creates ```model_metadata```

File: sample_configs/config_fasttext.json (hunk 7 above)
Comment (yfletberliac) [reply sample_configs/config_fasttext.json]: yes - also added ```"language": "fr"```

**Review by pocman — APPROVED** [review]

**Pull request merged**